hibike/README.md

```diff
@@ -46,75 +104,118 @@ Each Smart Device will be assigned an 88-bit UID with the following data.
+- Hibike abstracts every smart device as a set of parameters that map to values
+- Each smart device contains some number of paramaters, which can be read/written to.
 
-Year        - 8-bit ID corresponding to the competition year that the Smart Device was manufactured
-              for. The 2015-2016 season will correspond to 0x00
+- Paramaters can have many types, such as bool, int, short, byte, float, etc.
+- Some paramaters are read only, some are write only, and some support both.
+- A config file will describe the paramaters for each Device Type (name, type, permissions).
+- Some packets encode sets of parameters in the form of bitmaps.
 
-ID          - Randomly generated 64-bit number that will uniquely identify each Smart Device within
-              a specific device type and year. With 64-bit IDs, the probability of a hash collision 
-              with 1000 of 1 type of device per year is roughly 0.05%
 
-## Section 3: Enumerations
+
+
+    ```
+    +-------------------+-----------------------------+     +-----------------------------+
+    |      Params       |         Value 0             | ... |         Value 15            |
+    |     (16 bits)     |  (Optional and Variable)    |     |  (Optional and Variable)    |
+    +-------------------+-----------------------------+     +-----------------------------+
+    ```
+
+- Params - 16-bit bitmap describing a set of parameters. The nth bit of Params, where the LSB is the 0th bit, references the nth paramater of a device.
+
+- Value[0-15] - DeviceWrite and DeviceData send actual values for each param in Params. The value field for param n will only be present if bit n in Params is set. The size and type of each value field depends on param number and device type, and is described in a config file.
+
+
+## Section 4: Enumerations
 
 Message ID Enumeration:
 
-    +---------+--------------------------+
-    |   ID    |       Message Type       |
-    +------------------------------------+
-    |  0x00   |   Subscription Request   |
-    +------------------------------------+
-    |  0x01   |   Subscription Response  |
-    +------------------------------------+
-    |  0x02   |        Data Update       |
-    +------------------------------------+
-    |  0x03   |       Device Update      |
-    +------------------------------------+
-    |  0x04   |       Device Status      |
-    +------------------------------------+
-    |  0x05   |      Device Response     |
-    +------------------------------------+
-    |  0x06   |          Ping            |
-    +------------------------------------+
-    |  0x08   |    Description Request   |
-    +------------------------------------+
-    |  0x09   |   Description Response   |
-    +------------------------------------+
-    |  0xFF   |           Error          |
-    +------------------------------------+
+|   ID    |       Message Type       |
+|---------|--------------------------|
+|  0x10   |          Ping            |
```

Comment:
Have you thought carefully about whether future kits will be able to extend hibike in a backward-compatible way? Breaking changes this year is fine, but ideally we could use a few years of experience building hibike to then lock in a stable system.